lib/tooling/bronto-refactor-tool.ts

```diff
@@ -0,0 +1,70 @@
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import {splitArguments} from '../../shared/common-utils.js';
+import {CompilationInfo} from '../../types/compilation/compilation.interfaces.js';
+import {ToolInfo} from '../../types/tool.interfaces.js';
+import {OptionsHandlerLibrary} from '../options-handler.js';
+
+import {ToolEnv} from './base-tool.interface.js';
+import {BaseTool} from './base-tool.js';
+
+export class BrontoRefactorTool extends BaseTool {
+    static get key() {
+        return 'bronto-refactor-tool';
+    }
+
+    constructor(toolInfo: ToolInfo, env: ToolEnv) {
+        super(toolInfo, env);
+
+        this.addOptionsToToolArgs = false;
+    }
+
+    override async runTool(
+        compilationInfo: CompilationInfo,
+        inputFilepath: string,
+        args?: string[],
+        stdin?: string,
+        supportedLibraries?: Record<string, OptionsHandlerLibrary>,
+    ) {
+        const sourcefile = inputFilepath;
+        const options = compilationInfo.options;
+        const includeflags = super.getIncludeArguments(compilationInfo.libraries, supportedLibraries || {});
+        const libOptions = super.getLibraryOptions(compilationInfo.libraries, supportedLibraries || {});
+
+        let compileFlags = ['compiler-explorer', sourcefile, '--'].concat(
+            splitArguments(compilationInfo.compiler.options),
+        );
+        compileFlags = compileFlags.concat(includeflags);
+        compileFlags = compileFlags.concat(libOptions);
+
+        const manualCompileFlags = options.filter(option => option !== sourcefile);
+        compileFlags = compileFlags.concat(manualCompileFlags);
+
+        const result = await super.runTool(compilationInfo, sourcefile, compileFlags);
```

Comment:
Hi!

I installed the tool locally (and have a plan for installing it on the live site!)

I can't get this to work locally:
![image](https://github.com/user-attachments/assets/003127c3-ab93-4738-a3d8-bb8ce8ef1682)

The tool doesn't seem to do anything.

In the screenshot you seem to have put something in `stdin`, but in the PR you've disabled stdin?

I don't quite understand why it runs the tool twice? Once without stdin, and then once _with_ stdin?

You also seem to ignore the `args` ?

What might I be doing wrong here? :)